kbpagesconfig/acl.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/keybase/cli"
+)
+
+var aclSetAnonymousCmd = cli.Command{
+	Name:         "set-anonymous",
+	Usage:        "set anonymous permission(s) for the given path(s)",
+	ArgumentHelp: "set-anonymous <read|list|read,list> <path> [path ...]",
+	Action: func(c *cli.Context) {
+		if len(c.Args()) < 2 {
+			fmt.Fprintln(os.Stderr, "need at least 2 args")
+			os.Exit(1)
+		}
+		editor, err := newKBPConfigEditor(c.GlobalString("dir"))
+		if err != nil {
+			fmt.Fprintf(os.Stderr,
+				"creating config editor error: %v", err)
+			os.Exit(1)
+		}
+		for _, p := range c.Args()[1:] {
+			err := editor.setAnonymousPermission(c.Args()[0], p)
+			if err != nil {
+				fmt.Fprintf(os.Stderr,
+					"setting anonymous permission %q on %q error: %v\n",
+					c.Args()[0], p, err)
+				os.Exit(1)
+			}
+		}
+		if err := editor.confirmAndWrite(); err != nil {
+			fmt.Fprintf(os.Stderr, "writing new config error: %v\n", err)
+			os.Exit(1)
+		}
+	},
+}
+
+var aclClearCmd = cli.Command{
+	Name:         "clear",
+	Usage:        "clear the ACL for the given path(s)",
+	ArgumentHelp: "clear <path> [path ...]",
+	Action: func(c *cli.Context) {
+		if len(c.Args()) < 1 {
+			fmt.Fprintln(os.Stderr, "need at least 1 arg")
+			os.Exit(1)
+		}
+		editor, err := newKBPConfigEditor(c.GlobalString("dir"))
+		if err != nil {
+			fmt.Fprintf(os.Stderr,
+				"creating config editor error: %v", err)
+			os.Exit(1)
+		}
+		for _, p := range c.Args() {
+			editor.clearACL(p)
+		}
+		if err := editor.confirmAndWrite(); err != nil {
+			fmt.Fprintf(os.Stderr, "writing new config error: %v\n", err)
+			os.Exit(1)
+		}
+	},
+}
+
+var aclSetAdditionalCmd = cli.Command{
+	Name: "set-additional",
+	Usage: "set permission(s) <username> is granted on the " +
```

Comment:
I'm having trouble parsing this sentence.  Maybe it should be something like: "set a given user's permissions on the given path(s), overriding the permissions for an anonymous user".

Reply:
It doesn't override though; it's additional on top of anonymous ones. Would "set additional permission(s) that <username> is granted on top of anonymous ones on the given path(s) "

Reply:
"is granted" -> "are granted", but otherwise yes that's better thanks.